lib/scorecard.rb

```diff
@@ -0,0 +1,53 @@
+require 'frame.rb'
+
+class Scorecard
+
+  attr_accessor :frames, :total_score
+
+  def initialize(frames = Frame.new )
+    @frames = [frames]
+    @current_turn = 0
+    @total_score = 0
+  end
+
+  def roll(pins)
+    if @frames[turn].next_turn?
+      create_new_frame
+      @frames[turn].rolled(pins)
+      calculate_total
+    else
+      @frames[turn].rolled(pins)
+      calculate_total
+    end
+  end
+
+  def turn
+    @current_turn
+  end
+
+  def calculate_total
+    @total_score = 0
+    calcualate_bonus
+    @frames.each { |frame| @total_score += (frame.score + frame.bonus_score) }
```

Comment:
This method returns an array of frames currently - should it return 'total_score'?